File: poultry_pal_plus_app/lib/models/farm_report.dart
import 'coop_report.dart';

class FarmReport {
  final int totalChickens;
  final double totalSales;
  final double totalExpenses;
  final int totalMortalities;
  final List<CoopReport> coopReports;

  FarmReport({
    required this.totalChickens,
    required this.totalSales,
    required this.totalExpenses,
    required this.totalMortalities,
    required this.coopReports,
  });

  factory FarmReport.fromJson(Map<String, dynamic> json) {
    return FarmReport(
      totalChickens: json['totalChickens'],
      totalSales: json['totalSales'],
      totalExpenses: json['totalExpenses'],
      totalMortalities: json['totalMortalities'],
      coopReports: (json['coopReports'] as List).map((json) => CoopReport.fromJson(json)).toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'totalChickens': totalChickens,
      'totalSales': totalSales,
      'totalExpenses': totalExpenses,
      'totalMortalities': totalMortalities,
      'coopReports': coopReports.map((report) => report.toJson()).toList(),
    };
  }
}

File: poultry-pal-service/src/main/java/co/za/hlaluko/dynamics/poultry/pal/model/dto/FarmReport.java
package co.za.hlaluko.dynamics.poultry.pal.model.dto;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FarmReport {

  private int totalChickens;
  private double totalSales;
  private double totalExpenses;
  private int totalMortalities;
  private List<CoopReport> coopReports;

}
